src/subtype.c

```diff
@@ -387,6 +387,10 @@ static int var_gt(jl_tvar_t *b, jl_value_t *a, jl_stenv_t *e, int param)
         return subtype_ufirst(a, bb->lb, e);
     if (!((bb->ub == (jl_value_t*)jl_any_type && !jl_is_type(a) && !jl_is_typevar(a)) || subtype_ufirst(a, bb->ub, e)))
         return 0;
+    if (jl_is_typevar(bb->lb) && !jl_is_type(a) && bb->lb == bb->ub && var_gt((jl_tvar_t *) bb->lb, a, e, param))
+        return 1;
+    if (jl_is_typevar(a) && !jl_is_type(bb->lb) && bb->lb == bb->ub && var_lt((jl_tvar_t *) a, bb->lb, e, param))
```

Comment:
Since `var_gt` and `var_lt` can make state changes, it's better to return their result in this case, since otherwise you're left with state the algorithm is not using.